promql/functions.go

```diff
@@ -814,11 +863,11 @@ var functions = map[string]*Function{
 		Call:       funcResets,
 	},
 	"round": {
-		Name:         "round",
-		ArgTypes:     []model.ValueType{model.ValVector, model.ValScalar},
-		OptionalArgs: 1,
-		ReturnType:   model.ValVector,
-		Call:         funcRound,
+		Name:       "round",
+		ArgTypes:   []model.ValueType{model.ValVector, model.ValScalar},
+		Variadic:   true,
```

Comment:
Hmm, now I could reply with your own comment about this, "However, this hack would not be an acceptable solution to me." :) Functions that just silently ignore extra arguments sound uglier to me than supporting both variadic and optional args. (Python goes the full way here, regular args, optional args, variadic args, and named args, but I wouldn't go that far in Prometheus).

If we don't want to have both *concepts* (variadic and optional) around, an alternative could be to just have variadic args, but being able to limit them to a certain number (like 1) to emulate optional args?